File: Codesignal/elixir/adjacente_elements_product.ex
def solution(inputArray) do
  calc(inputArray, [])
end

def calc([head | tail], results) do
 case Enum.count(tail) > 0 do
   true -> get_product([head | tail], results)
   false -> Enum.max(results)
 end
end

defp get_product([head | tail], results) do
 value = head * List.first(tail)
 new_results = [value | results]
 calc(tail, new_results)
end

File: Codesignal/js/adjacente_elements_product.js
function solution(inputArray) {
    const multiply = (a, b) => a * b;
    
  let biggest = multiply(inputArray[0], inputArray[1]);

  for (let i = 0; i < inputArray.length; i++) {
    let next = inputArray[i + 1];

    if (multiply(inputArray[i], next) > biggest) {
      biggest = multiply(inputArray[i], next);
    }
  }
  return biggest;
}